File: OCSimpleXIB/OCSimpleXIB/ViewController.m
//
//  ViewController.m
//  OCSimpleXIB
//
//  Created by Don Mag on 10/4/17.
//  Copyright © 2017 DonMag. All rights reserved.
//

#import "ViewController.h"

#import "OrangeView.h"
#import "GreenView.h"

@interface ViewController () {
	OrangeView *ov;
	GreenView *gv;
}

@property (weak, nonatomic) IBOutlet UIView *viewOrange;
@property (weak, nonatomic) IBOutlet UIView *viewGreen;

@end

@implementation ViewController

- (void)viewDidLoad {
	[super viewDidLoad];
	// Do any additional setup after loading the view, typically from a nib.
	
	ov = [OrangeView new];
	gv = [GreenView new];
	
	ov.translatesAutoresizingMaskIntoConstraints = false;
	gv.translatesAutoresizingMaskIntoConstraints = false;
	
	[_viewOrange addSubview:ov];
	[_viewGreen addSubview:gv];

	[ov.topAnchor constraintEqualToAnchor:_viewOrange.topAnchor constant:0.0].active = YES;
	[ov.bottomAnchor constraintEqualToAnchor:_viewOrange.bottomAnchor constant:0.0].active = YES;
	[ov.leadingAnchor constraintEqualToAnchor:_viewOrange.leadingAnchor constant:0.0].active = YES;
	[ov.trailingAnchor constraintEqualToAnchor:_viewOrange.trailingAnchor constant:0.0].active = YES;
	
	[gv.topAnchor constraintEqualToAnchor:_viewGreen.topAnchor constant:0.0].active = YES;
	[gv.bottomAnchor constraintEqualToAnchor:_viewGreen.bottomAnchor constant:0.0].active = YES;
	[gv.leadingAnchor constraintEqualToAnchor:_viewGreen.leadingAnchor constant:0.0].active = YES;
	[gv.trailingAnchor constraintEqualToAnchor:_viewGreen.trailingAnchor constant:0.0].active = YES;
	
}


- (void)didReceiveMemoryWarning {
	[super didReceiveMemoryWarning];
	// Dispose of any resources that can be recreated.
}


@end

File: SimpleXIB/SimpleXIB/ViewController.swift
//
//  ViewController.swift
//  SimpleXIB
//
//  Created by Don Mag on 8/7/17.
//  Copyright © 2017 DonMag. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
	
	var theXIBView: XIBViewView?

	
	@IBAction func doChangeXIBLabel(_ sender: Any) {
		theXIBView?.setMeUp("Here's a new string!")
	}
	
	override func viewDidLoad() {
		super.viewDidLoad()
		// Do any additional setup after loading the view, typically from a nib.
		
		if let xibView = Bundle.main.loadNibNamed("XIBView", owner: nil, options: nil)?.first as? XIBViewView {
			
			xibView.finishSetup()
			
			xibView.translatesAutoresizingMaskIntoConstraints = false
			
			self.view.addSubview(xibView)
			
			xibView.centerXAnchor.constraint(equalTo: self.view.centerXAnchor).isActive = true
			xibView.centerYAnchor.constraint(equalTo: self.view.centerYAnchor).isActive = true
			
			xibView.widthAnchor.constraint(equalTo: self.view.widthAnchor, multiplier: 0.9).isActive = true
			xibView.heightAnchor.constraint(equalTo: xibView.widthAnchor, multiplier: 0.5).isActive = true
			
			self.theXIBView = xibView
			
		}
		
	}


}

